---
title: "MS_analysis_MQ_summary"
author: "Sam Siljee"
date: '2022-08-22'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Packages
```{r packages, include=FALSE}
library("tidyverse")
library("seqinr")
```

## Importing raw data
```{r reading in MQ results from the combined folder, include=FALSE}
# importing evidence file, this is the file suggested as the main analysis file
evidence <- read_tsv("~/Coding/MS_analysis/data/combined/txt/evidence.txt") %>%
  as.data.frame()
# removing decoys
evidence <- evidence[-which(evidence$Reverse == "+"),]
# adding column with first protein name if multiple protein matches
evidence$firstProtName <- sapply(str_split(evidence$`Protein names`,";",),'[',1)
# doing the same for multiple gene name matches
evidence$firstGeneName <- sapply(str_split(evidence$`Gene names`,";",),'[',1)
# converting the "Potential contaminant" column to a vector. The "Potential contaminant" column is unhelpfully laid out, with contaminants labeled as "+". and non-contaminants labeled as NA. Hence the slightly awkward phrasing of the below code, which takes the inverse of logical return from `is.na`
evidence$potCon <- !is.na(evidence$`Potential contaminant`)

# importing all peptides
allPeptides <- read_tsv("~/Coding/MS_analysis/data/combined/txt/allPeptides.txt") %>%
  as.data.frame()

# importing phospho sites
phosphoSites <- read_tsv("~/Coding/MS_analysis/data/combined/txt/Phospho (STY)Sites.txt") %>%
  as.data.frame()
# removing decoys
phosphoSites <- phosphoSites[-which(phosphoSites$Reverse == "+"),]
# adding column with first protein name if peptide matches multiple proteins
phosphoSites$firstProtName <- sapply(str_split(phosphoSites$`Protein names`,";",),'[',1)

# extracting basal cell unique markers list from "The Human Airway Epithelial Basal Cell Transcriptome, Hacket et al. PLoS ONE, 2011). Adding p63 and Keratin 5 manually, and adding some CD markers identified in Laar et al 2014.
basalCellGenes <- read.csv("~/Coding/MS_analysis/tracked_files/basalCellTranscriptome.csv") %>%
  .$Gene.symbol %>% 
  c(., "TP63", "KRT5", "FAS", "NT5E", "EPCAM", "TSPAN4", "F3", "NGFR", "IFNGR1", "MST1R", "IGF1R", "IGF2R", "ERBB2", "EGFR") %>% 
  unique() %>%
  sort()

#Adding comparison for cilia related genes. taken from http://www.syscilia.org/goldstandard.shtml and edited to remove all unrelated columns
sysciliaCiliatedCellGenes <- read.table("~/Coding/MS_analysis/tracked_files/scgs.v1.tsv", sep = "\t") %>%
  .$V1
#Adding list of ciliated genes taken from "The transcriptional signature associated with human motile cilia", Patir et al. 2020 Nature. Table S2 edited to include only lung genes, pseudogenes, novel transcripts, unvalidated genes removed, and saved as .csv for import to R.  https://www.nature.com/articles/s41598-020-66453-4#MOESM1
patirCiliatedCellGenes <- read.csv("~/Coding/MS_analysis/tracked_files/patir_cil_genes.csv") %>%
  .$Lung
#combining the above lists
ciliatedCellGenes <- c(sysciliaCiliatedCellGenes, patirCiliatedCellGenes) %>% 
  unique() %>% 
  sort()

#creating a list of all proteins from original fasta database
fasta <- read.fasta("~/Coding/MS_analysis/data/UP000005640_9606.fasta", seqtype = "AA", as.string = T)
```

## Enrichment for phosphopeptides
What proportion of results are enriched for phospho groups

```{r Phospho enrichment, echo=FALSE}
print("Fraction of results with at least one phospho modification") 
mean(evidence$`Phospho (STY)` > 0)
```

# Identifying basal cell markers
Listing markers found in phosphoproteome, along with fraction of markers found
```{r identifying basal cell markers, echo=FALSE}
markersPresent <- evidence %>%
  filter(firstGeneName %in% basalCellGenes) %>%
  .$firstProtName %>%
  unique() %>%
  sort()
print("Basal cell markers found")
markersPresent

print("Proportion of basal cell markers found")
length(markersPresent)/length(basalCellGenes)
```

# Identifying ciliated cell markers
Listing markers found in phosphoproteome, along with fraction of markers found
```{r identifying ciliated cell markers, echo=FALSE}
#first looking at the overlapping genes found in both the basal cell and ciliated cell sets
print("Genes found in both basal cell and ciliated cell gene list")
intersect(basalCellGenes, ciliatedCellGenes)

ciliatedMarkersPresent <- evidence %>%
  filter(firstGeneName %in% ciliatedCellGenes) %>%
  .$firstProtName %>%
  unique() %>%
  sort()
print("Ciliated cell markers found")
ciliatedMarkersPresent

print("Proportion of ciliated cell markers found - total list (689)")
length(ciliatedMarkersPresent)/length(ciliatedCellGenes)
```

## Contaminants

# Looking at fraction of contaminants
Dividing the entries in contaminants marked as "potential contaminant" by total number of entries

```{r Summarising fraction of contaminants, echo=FALSE}
print("Fraction of results categorised as contaminants")
length(which(evidence$`Potential contaminant` == "+")) / nrow(evidence)
```

# Listing contaminants
Listing all the potential contaminants from the phospho sites data, note that at least Keratin 5 is found in basal cells, and is therefore expected

```{r looking at the identified contaminants, echo=FALSE}
#contaminants in the phosphoSites data, first creating a string of all the protein names
print("List of contaminants found")
evidence %>%
  filter(potCon == TRUE) %>%
  .$firstProtName %>%
  unique() %>%
  sort()
```
